From 8490f0961093364d2b0d218393ac8a3f2668f641 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Tue, 16 Apr 2024 10:52:00 +0000
Subject: [PATCH] GCC: fix template method call in https_record_rdata.cc
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

GCC does not allow to call a template method with its parameters.
It shows a warning first: "expected ‘template’ keyword before
dependent template name". But then it fails to parse properly.

Bug: 40565911
Change-Id: I0faeec539d141eae33854ac53f5351b9697c0039
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5456356
Commit-Queue: José Dapena Paz <jdapena@igalia.com>
Reviewed-by: Tsuyoshi Horo <horo@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1287934}
---
 net/dns/https_record_rdata.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/dns/https_record_rdata.cc b/net/dns/https_record_rdata.cc
index e534f9aa13e..d23a51da13d 100644
--- a/net/dns/https_record_rdata.cc
+++ b/net/dns/https_record_rdata.cc
@@ -116,7 +116,7 @@ bool ParseIpAddresses(std::string_view param_value,
 
   std::vector<IPAddress> addresses;
   do {
-    if (auto addr_bytes = reader.Read<ADDRESS_SIZE>();
+    if (auto addr_bytes = reader.template Read<ADDRESS_SIZE>();
         !addr_bytes.has_value()) {
       return false;
     } else {
